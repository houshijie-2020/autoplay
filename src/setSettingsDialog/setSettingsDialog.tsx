import { Notebook } from '@jupyterlab/notebook';
import { Widget } from '@lumino/widgets';
import React from 'react';
import ReactDOM from 'react-dom';
import { getMetadata } from '../notebookActions';
import { AutoplayMetadata, DEFAULT_AUTOPLAY_METADATA } from '../types.model';

/**
 * A widget used to display the options so the user can setup the notebook
 */
export class SetSettingsDialog extends Widget {
    metadata: AutoplayMetadata;

    constructor(notebook: Notebook) {
        const wrapper = document.createElement('div');
        wrapper.className = "ap-setSettingsDialog";
        super({ node: wrapper });
        this.metadata = getMetadata(notebook) || DEFAULT_AUTOPLAY_METADATA;

        const content = SetSettingsContent({ metadata: this.metadata });
        ReactDOM.render(content, wrapper);
    }

    /**
     * Get the value of the widget.
     */
    getValue(): AutoplayMetadata {
        return this.metadata;
    }
}

interface ISetSettingsContent {
    metadata: AutoplayMetadata
}

const SetSettingsContent = ({ metadata }: ISetSettingsContent) => (
    <div className="ap-setSettingsDialog-form">
        <label>
            <input type="checkbox"
                defaultChecked={metadata.autoRun}
                onChange={() => metadata.autoRun = !metadata.autoRun} />
                Run all cells
        </label>
        <label>
            <input type="checkbox"
                defaultChecked={metadata.hideCodeCells}
                onChange={() => metadata.hideCodeCells = !metadata.hideCodeCells} />
                Hide all code cells
        </label>
    </div>
);

export { SetSettingsContent };

