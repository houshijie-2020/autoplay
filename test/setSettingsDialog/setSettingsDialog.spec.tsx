import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import 'ts-jest';
import { SetSettingsContent } from '../../src/setSettingsDialog/setSettingsDialog';
import { AutoplayMetadata } from '../../src/types.model';

jest.setTimeout(10000);

const autoplayMetadata: AutoplayMetadata = {
    "autoRun": true,
    "hideCodeCells": true
}

describe('SetSettingDialog', () => {

    it('should show popup content', () => {
        const renderer = ShallowRenderer.createRenderer();

        const component = renderer.render(
            <SetSettingsContent metadata={autoplayMetadata}></SetSettingsContent>
        );

        expect(component).toMatchSnapshot();
    });
});
