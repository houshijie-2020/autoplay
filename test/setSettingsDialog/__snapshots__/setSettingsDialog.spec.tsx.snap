// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetSettingDialog should show popup content 1`] = `
<div
  className="ap-setSettingsDialog-form"
>
  <label>
    <input
      defaultChecked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Run all cells
  </label>
  <label>
    <input
      defaultChecked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Hide all code cells
  </label>
</div>
`;
